test/parallel/test-stringbytes-external-exceed-max-by-1.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+// v8 fails silently if string length > v8::String::kMaxLength
+// v8::String::kMaxLength defined in v8.h
+const kStringMaxLength = process.binding('buffer').kStringMaxLength;
+
+try {
+  new Buffer(kStringMaxLength * 3);
+} catch(e) {
+  assert.equal(e.message, 'Invalid array buffer length');
+  console.log(
+      '1..0 # Skipped: intensive toString tests due to memory confinements');
+  return;
+}
+
+const buf1 = new Buffer(kStringMaxLength + 1);
+
+assert.throws(function() {
+  buf1.toString();
+}, /toString failed|Invalid array buffer length/);
+
+assert.throws(function() {
+  buf1.toString('ascii');
+}, /toString failed/);
+
+assert.throws(function() {
+  buf1.toString('utf8');
+}, /toString failed/);
```

Comment:
Interestingly, this will still throw as expected even if there are multibyte characters in the string. It must have something to do with v8 internals.

On the side, I don't think that `utf8`, `hex` and `base64` are necessary. `utf8` for reasons similar to the above. `hex` and `base64` will never be shorter than `binary`. It may be useful to add `ucs2`, but not sure if doing the larger allocation will break the test on pi.